gbpservice/nfp/configurator/drivers/loadbalancer/v2/haproxy/haproxy_driver.py

```diff
@@ -0,0 +1,592 @@
+
+    def update(self, context, old_obj, obj):
+        LOG.info(_LI("LB %(cls_name)s, update %(id)s"),
+                 {"cls_name": self.__class__.__name__, "id": obj['id']})
+
+    def delete(self, context, obj):
+        LOG.info(_LI("LB %(cls_name)s, delete %(id)s"),
+                 {"cls_name": self.__class__.__name__, "id": obj['id']})
+
+
+class HaproxyLoadBalancerManager(HaproxyCommonManager,
+                                 n_driver_base.BaseLoadBalancerManager):
+
+    def _get_amphorae_network_config(self,
+                                     context,
+                                     loadbalancer_dict,
+                                     loadbalancer_o_obj):
+        loadbalancer_n_obj = n_data_models.LoadBalancer.from_dict(
+            copy.deepcopy(loadbalancer_dict))
+
+        amphorae_network_config = {}
+
+        for amp in loadbalancer_o_obj.amphorae:
+            if amp.status != constants.DELETED:
+                # Get vip_subnet
+                vip_subnet = None
+                for subnet_dict in context['service_info']['subnets']:
+                    if subnet_dict['id'] == loadbalancer_n_obj.vip_subnet_id:
+                        vip_subnet = n_data_models.Subnet.from_dict(
+                            copy.deepcopy(subnet_dict))
+                        break
+                if vip_subnet is None:
+                    raise exceptions.IncompleteData(
+                        "VIP subnet information is not found")
+
+                sc_metadata = ast.literal_eval(
+                    loadbalancer_dict['description'])
+                vrrp_port = n_data_models.Port(
+                    mac_address=sc_metadata['provider_interface_mac'])
+                if vrrp_port is None:
+                    raise exceptions.IncompleteData(
+                        "VRRP port information is not found")
+
+                amphorae_network_config[amp.id] = \
+                    network_data_models.AmphoraNetworkConfig(
+                        amphora=amp,
+                        vip_subnet=vip_subnet,
+                        vrrp_port=vrrp_port
+                    )
+
+        return amphorae_network_config
+
+    def create(self, context, loadbalancer):
+        LOG.info(_LI("LB %(cls_name)s, create %(id)s"),
+                 {"cls_name": self.__class__.__name__,
+                  "id": loadbalancer['id']})
+
+        self.driver.add_amphora(loadbalancer['id'],
+                                loadbalancer['description'])
+        loadbalancer_o_obj = self.driver.o_models_builder.\
+            get_loadbalancer_octavia_model(loadbalancer)
+        amphorae_network_config = self._get_amphorae_network_config(
+                                     context, loadbalancer, loadbalancer_o_obj)
+        self.driver.amphora_driver.post_vip_plug(
+                loadbalancer_o_obj, amphorae_network_config)
+        LOG.info(_LI("Notfied amphora of vip plug"))
```

Comment:
Please add more info like resource id to logs wherever possible. It helps in debugging.